providers/implementations/signature/ecdsa_sig.c

```diff
@@ -141,9 +164,103 @@ static void *ecdsa_newctx(void *provctx, const char *propq)
     return ctx;
 }
 
-static int ecdsa_signverify_init(PROV_ECDSA_CTX *ctx, void *ec,
-                                 const OSSL_PARAM params[], int operation,
-                                 const char *desc)
+static int ecdsa_setup_md(PROV_ECDSA_CTX *ctx,
+                          const char *mdname, const char *mdprops,
+                          const char *desc)
+{
+    EVP_MD *md = NULL;
+    size_t mdname_len;
+    int md_nid, md_size;
+    WPACKET pkt;
+
+    if (mdname == NULL)
+        return 1;
+
+    mdname_len = strlen(mdname);
+    if (mdname_len >= sizeof(ctx->mdname)) {
+        ERR_raise_data(ERR_LIB_PROV, PROV_R_INVALID_DIGEST,
+                       "%s exceeds name buffer length", mdname);
+        return 0;
+    }
+    if (mdprops == NULL)
+        mdprops = ctx->propq;
+    md = EVP_MD_fetch(ctx->libctx, mdname, mdprops);
+    if (md == NULL) {
+        ERR_raise_data(ERR_LIB_PROV, PROV_R_INVALID_DIGEST,
+                       "%s could not be fetched", mdname);
+        return 0;
+    }
+    md_size = EVP_MD_get_size(md);
+    if (md_size <= 0) {
+        ERR_raise_data(ERR_LIB_PROV, PROV_R_INVALID_DIGEST,
+                       "%s has invalid md size %d", mdname, md_size);
+        goto err;
+    }
+    md_nid = ossl_digest_get_approved_nid(md);
+    if (md_nid < 0) {
```

Comment:
`ossl_digest_get_approved_nid` seems to return `NID_undef` if the digest is not allowed. But `NID_undef` has a value of 0, so the `< 0` here should probably be `<= 0`. Or just `== NID_undef`.

There's a similar check in `dsa_setup_md` which also looks wrong to me (but this PR didn't change it)

Reply:
I wonder why the code looked like this comming in, then...